src/marks/image.js

```diff
@@ -40,17 +40,19 @@ function maybePathChannel(value) {
 
 export class Image extends Mark {
   constructor(data, options = {}) {
-    let {x, y, width, height, src, preserveAspectRatio, crossOrigin, frameAnchor, imageRendering} = options;
+    let {x, y, r, width, height, src, preserveAspectRatio, crossOrigin, frameAnchor, imageRendering} = options;
     if (width === undefined && height !== undefined) width = height;
     else if (height === undefined && width !== undefined) height = width;
     const [vs, cs] = maybePathChannel(src);
-    const [vw, cw] = maybeNumberChannel(width, 16);
-    const [vh, ch] = maybeNumberChannel(height, 16);
+    const [vr, cr] = maybeNumberChannel(r);
+    const [vw, cw] = maybeNumberChannel(width, r != null ? cr : 16);
+    const [vh, ch] = maybeNumberChannel(height, r != null ? cr : 16);
```

Comment:
Slight preference for treating a null `r` as zero instead of undefined, so that we don’t have to accommodate the null type (and instead it is implicitly coerced to a number). Also, I’d prefer if this logic shook out like this, instead of needing new this.sw, this.sh, this.clipR fields:

```suggestion
    const [vw, cw] = maybeNumberChannel(width, r === undefined ? 16 : cr * 2);
    const [vh, ch] = maybeNumberChannel(height, r === undefined ? 16 : cr * 2);
```